### PR TITLE
feat: migration to task processing api

### PR DESCRIPTION
**Pull request by kyteinsky**
All related PRs:
https://github.com/cloud-py-api/nc_py_api/pull/284
https://github.com/cloud-py-api/app_api/pull/359
https://github.com/nextcloud/translate2/pull/11

### CODE REVIEW

File: lib/main.py [hunk 1]
```diff
@@ -45,33 +45,45 @@ def __setitem__(self, key, value):
...
 service = Service(config)
 
 
+def report_error(task: dict | None, exc: Exception):
+    with suppress(Exception):
+        nc = NextcloudApp()
+        nc.log(LogLvl.ERROR, str(exc))
+        if task:
+            nc.providers.task_processing.report_result(task["id"], error_message=str(exc))
+
```
Comment (marcelklehr) [inline lib/main.py:79]: Nice to extract this!

**Review by marcelklehr — APPROVED** [review]
Looks very clean! :tada:

**Review by bigcat88 — APPROVED** [review]

File: lib/main.py [hunk 2]
```diff
@@ -80,62 +92,86 @@ async def _(request: Request, exc: Exception):
...
+        task_type=TASK_TYPE_ID,
+        input_shape_enum_values={
+            "origin_language": [DETECT_LANGUAGE],
+            "target_language": languages,
+        },
+        input_shape_defaults={
+            "origin_language": DETECT_LANGUAGE.value,
+        },
+    )
+    await nc.providers.task_processing.register(provider)
```
Comment (marcelklehr) [inline lib/main.py:160]: This will fail now, I think, as app_api now wants all possible keys? Or does the python implementation set defaults?

Comment (bigcat88) [reply lib/main.py:160]: python will set all by defaults

**Pull request merged**